src/orange/orangeinp/ConvexRegion.cc

```diff
@@ -309,6 +310,88 @@ void Ellipsoid::output(JsonPimpl* j) const
     to_json_pimpl(j, *this);
 }
 
+//---------------------------------------------------------------------------//
+// GENTRAP
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from half Z height and 1-4 vertices for top and bottom planes
+ */
+GenTrap::GenTrap(real_type halfz, VecReal2 const& lo, VecReal2 const& hi)
+    : hz_{halfz}, lo_{lo}, hi_{hi}
```

Comment:
Why don't we capture and move these to avoid the extra allocation:
```suggestion
GenTrap::GenTrap(real_type halfz, VecReal2&& lo, VecReal2&& hi)
    : hz_{halfz}, lo_{std::move(lo)}, hi_{std::move(hi)}
```